gitlab/resource_gitlab_project.go

```diff
@@ -574,6 +588,14 @@ func resourceGitlabProjectUpdate(d *schema.ResourceData, meta interface{}) error
 		options.PagesAccessLevel = stringToAccessControlValue(d.Get("pages_access_level").(string))
 	}
 
+	if d.HasChange("mirror") {
+		options.Mirror = gitlab.Bool(d.Get("mirror").(bool))
+	}
+
+	if d.HasChange("mirror_trigger_builds") {
+		options.MirrorTriggerBuilds = gitlab.Bool(d.Get("mirror_trigger_builds").(bool))
+	}
```

Comment:
When I run the tests locally, the create test is passing but the update test is failing. 

I've done some debugging and I don't think this is an issue with the Terraform Provider. When I curl my GitLab api with the params `{ mirror: false, mirror_trigger_builds: false }`, the api will also error out. 

The GitLab [api tests](https://gitlab.com/gitlab-org/gitlab/-/blob/master/ee/spec/requests/api/projects_spec.rb#L798) seem to always include the `import_url` field with their params.  I'll have to do more research to see if this is a bug, or if this was intentional. 

If your GitLab api is also failing, you may need to change this to something like:

```
if d.HasChange("mirror") {
  options.Mirror = gitlab.Bool(d.Get("mirror").(bool))
  options.ImportURL = gitlab.String(d.Get("import_url").(string))
}

if d.HasChange("mirror_trigger_builds") {
  options.MirrorTriggerBuilds = gitlab.Bool(d.Get("mirror_trigger_builds").(bool))
  options.ImportURL = gitlab.String(d.Get("import_url").(string))
}
```

Reply:
@btyy77c thank you for your awesome investigative work, i'll try your suggestion and report back!